game_engine.py

```diff
@@ -52,71 +52,83 @@ def setup_rounds(self, players, current_board, total_rounds=2):
 
     def take_turn(self, players, current_board):
         """
+        Take a turn (Roll Dice, Get Resources, Make a Decision, Do Decision)
         """
         for player in players:
             current_board.current_roll = GameEngine.roll_dice()
+            print("Dice Roll: " + str(current_board.current_roll))
+            
             self.give_resources_by_dice_roll(current_board, players, current_board.current_roll)
-            decision, vertex = self.evaluate_decision(player, current_board)
-            current_board = self.do_decision(player, decision, current_board, vertex)
+
+            print("Current Resources: " + str(player.resources))
+            
+            decision, vertex, trade_g, trade_f = self.evaluate_decision(player, current_board)
+            print("Decision made: " + str(decision) + "\n")
+
+            current_board = self.do_decision(player, decision, current_board, vertex, trade_g, trade_f)
 
         return current_board
 
     def evaluate_decision(self, player, current_board):
         """
+        Evaluate all possible decisions based on weighted heuristics
         """
         decisions = ["do_nothing", "build_settlement", "build_city", "build_road", "draw_development", "trade"]
         decision = "do_nothing"
         vertex = None
+        trade_get = None #What to trade to receive
+        trade_from = None #What to trade from
         highest_score = 0.5
         
         build_city_scores = self.calculate_city_scores(player, current_board)
         build_settlement_scores = self.calculate_settlement_scores(player, current_board)
         build_development_score = self.calculate_development_score(player, current_board)
-        trade_score = self.calculate_trade_score(player, current_board)
+        trade_score, trade_g, trade_f = self.calculate_trade_score(player, current_board)
         
         for idx, score in enumerate(build_city_scores):
             if score > highest_score:
                 highest_score = score
                 vertex = current_board.vertices[idx]
+                trade_get = None
+                trade_from = None
                 decision = "build_city"
         
         for idx, score in enumerate(build_settlement_scores):
             if score > highest_score:
                 highest_score = score
                 vertex = current_board.vertices[idx]
+                trade_get = None
+                trade_from = None
                 decision = "build_settlement"
```

Comment:
Can probably do a bunch of refactoring here due to some code duplication, but not highest priority.